src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,495 @@
+
+You can confirm that the node is still live and available using [`cockroach node status`]({% link {{ page.version.version }}/cockroach-node.md %}#node-status).
+
+### 3. WAL Failover during long disk stalls
+
+#### Cause long disk stall
+
+Before triggering a WAL Failover, open the [DB Console]({% link {{ page.version.version }}/ui-overview.md %}) so that we can observe WAL Failover metrics.
+
+Next, we will trigger WAL Failover by introducing a transient disk stall that is `100s` in duration (longer than the value of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables)).
+
+Create a file named `long-stall.sh` with the following content which will cause long disk stalls:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+while (true)
+do
+    date
+    echo "Stalling for 100 seconds"
+    sudo vim -s cmd.txt -es stall.txt
+    sleep 100
+    date
+    echo "Unstalling"
+    sudo vim -s cmd.txt -es unstall.txt
+    sleep 10
+    date
+    exit
+done
+~~~
+
+Next, make the script file executable:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+chmod 777 wal-flip.sh
+~~~
+
+Next, cause a long stall:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+long-stall.sh
+~~~
+
+#### WAL Failover metrics
+
+{% include {{ page.version.version }}/wal-failover-metrics.md %}
+
+Whenever a WAL failover occurs on a disk, `wal.failover.switch.count` for the associated store will increment by 1.
+
+Notice there is a switchover followed by the disk stall. Since the disk continues to be stalled for longer than the duration of [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables), the node goes down, and there is no more metrics data coming from that node.
+
+<img src="{{ 'images/v24.3/wal-failover-long-stall-metrics.jpg' | relative_url }}" alt="WAL Failover Metrics - Long Stall" style="border:1px solid #eee;max-width:100%" />
+
+You can confirm that the node is down (no longer live and available) using [`cockroach node status`]({% link {{ page.version.version }}/cockroach-node.md %}#node-status).
+
+## Summary
+
+### Important environment variables
+
+| Variable                                     | Description                                                                                                              | Default | Recommended value with WAL failover enabled |
+|----------------------------------------------+--------------------------------------------------------------------------------------------------------------------------+---------+---------------------------------------------|
+| `COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT` | The threshold above which an observed engine sync duration triggers a fatal error.                                       | `20s`   | `40s`                                       |
```

Comment:
Note that this environment variable is the _default_ for the `storage.max_sync_duration` cluster setting. If that cluster setting is explicitly set, it takes precedence over this environment variable.

Reply:
thanks, i added that info to this description of the env variable